runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/options/DataflowWorkerLoggingOptions.java

```diff
@@ -38,7 +38,47 @@
    * The set of log levels that can be used on the Dataflow worker.
    */
   public enum Level {
-    DEBUG, ERROR, INFO, TRACE, WARN
+    /**
+     * Special level used to turn off logging.
+     *
+     * @see java.util.logging.Level#OFF
```

Comment:
Both Log4j and Log4j 2, Logback, ... use decimal dotted named hierarchies so this is not just a JUL concept.
https://logging.apache.org/log4j/2.0/manual/architecture.html
http://logback.qos.ch/manual/architecture.html

The names of the levels that we have map much more easily on to the ones exposed within SLF4J, Log4j, JCL, ... then they do in JUL.

I would say we have a custom implementation of log level filtering on top of SLF4J and its an implementation note that we use JUL.